### PR TITLE
Feature/159 Ignore unsupported files inside policy directories

### LINKED ISSUE #159: Ignore unsupported files inside policy directories
Non-policy files placed into policy directories are failing the build. Some OSs (macOS) place some files in every directory and we cannot modify that.
We should ignore all unknown file types and log that we are doing this.

```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':build-bundle'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
        at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:277)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:262)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: com.ca.apim.gateway.cagatewayconfig.config.loader.policy.PolicyConverterException: Unknown policy file extension for file: .DS_Store
        at com.ca.apim.gateway.cagatewayconfig.config.loader.policy.PolicyConverterRegistry.lambda$getConverterFromFileName$1(PolicyConverterRegistry.java:31)
        at com.ca.apim.gateway.cagatewayconfig.config.loader.policy.PolicyConverterRegistry.getConverterFromFileName(PolicyConverterRegistry.java:31)
        at com.ca.apim.gateway.cagatewayconfig.config.loader.PolicyAndFolderLoader.loadPolicy(PolicyAndFolderLoader.java:79)
        at com.ca.apim.gateway.cagatewayconfig.config.loader.PolicyAndFolderLoader.loadPolicies(PolicyAndFolderLoader.java:68)
        at com.ca.apim.gateway.cagatewayconfig.config.loader.PolicyAndFolderLoader.load(PolicyAndFolderLoader.java:46)
        at com.ca.apim.gateway.cagatewayconfig.BundleFileBuilder.lambda$buildBundle$0(BundleFileBuilder.java:60)
        at com.ca.apim.gateway.cagatewayconfig.BundleFileBuilder.buildBundle(BundleFileBuilder.java:60)
        at com.ca.apim.gateway.cagatewayconfig.BuildDeploymentBundleTask.perform(BuildDeploymentBundleTask.java:56)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:801)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:768)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        ... 31 more
```

### PR DESCRIPTION
**Pull request by J-Lou**
Fixes #159 

## Proposed Changes
* Skip loading policy if extension is unsupported.

### CODE REVIEW

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/policy/PolicyConverterRegistry.java [hunk 1]
```diff
@@ -12,18 +12,34 @@
...
     public PolicyConverterRegistry(final Set<PolicyConverter> converters) {
         this.policyConverters = unmodifiableSet(converters);
     }
 
+    @NotNull
+    public boolean isValidPolicyExtension(String fileName) {
+        boolean isValidPolicyExtension = policyConverters.stream()
+            .anyMatch(converter -> fileName.endsWith(converter.getPolicyTypeExtension()));
+
+        if (isValidPolicyExtension) {
```
Comment (joao-borges) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/policy/PolicyConverterRegistry.java]: really minor, but you can skip the else if you do something like this:
```
if (!isValidPolicyExtension) {
    Log here
}

return isValidPolicyExtension;
```

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/policy/PolicyConverterRegistry.java [hunk 2]
```diff
@@ -12,18 +12,34 @@
...
+            return true;
+        } else {
+            LOGGER.log(Level.FINE, "Unknown policy file extension for file: " + fileName);
```
Comment (joao-borges) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/policy/PolicyConverterRegistry.java]: use WARNING level to make sure user see that weird things are located into policy directories

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/policy/PolicyConverterRegistry.java (hunk 1 above)
Comment (J-Lou) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/policy/PolicyConverterRegistry.java]: Agreed

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/policy/PolicyConverterRegistry.java [hunk 3]
```diff
@@ -12,18 +12,33 @@
...
 public class PolicyConverterRegistry {
     private final Set<PolicyConverter> policyConverters;
+    private static final Logger LOGGER = Logger.getLogger(PolicyConverterRegistry.class.getName());
 
     @Inject
     public PolicyConverterRegistry(final Set<PolicyConverter> converters) {
         this.policyConverters = unmodifiableSet(converters);
     }
 
+    @NotNull
```
Comment (kevpak) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/policy/PolicyConverterRegistry.java]: Remove this annotation. `boolean` is a primitive type and never null.

**Pull request merged**